File: LudoAPI/Models/Tiles/ArrowTile.cs
namespace LudoAPI.Models.Tiles;

public class ArrowTile : Tile
{
    private Move arrowMove;

    //used in UI
    public ArrowDirection ArrowDirection { get; }


    public ArrowTile(Coordinate coordinate, Color color, Move move, Move arrowMove) : base(coordinate, color, move)
    {
        this.arrowMove = arrowMove;
        ArrowDirection = CalculateArrowDirection();
    }

    private ArrowDirection CalculateArrowDirection()
    {
        var yChange = arrowMove.YChange;

        // todo make prettier
        switch (arrowMove.XChange)
        {
            case -1:
                if (yChange == 0) return ArrowDirection.Left;
                break;
            case 1:
                if (yChange == 0) return ArrowDirection.Right;
                break;
            case 0:
                switch (yChange)
                {
                    case 1:
                        return ArrowDirection.Up;
                    case -1:
                        return ArrowDirection.Down;
                }

                break;
        }

        throw new NotImplementedException("ArrowDirection is not supported");
    }

    public override Move NextMove(Piece piece)
    {
        throw new NotImplementedException();
    }
}

public enum ArrowDirection
{
    Up,
    Right,
    Down,
    Left,
}

File: LudoUI/models/ArrowTile.py
from enum import Enum

from PlayerColor import PlayerColor
from models.Coordinate import Coordinate
from models.Tile import Tile


class ArrowDirection(Enum):
    Up = 0
    Right = 1
    Down = 2
    Left = 3


class ArrowTile(Tile):

    def __init__(self, coordinate: Coordinate, color: PlayerColor, arrow_direction: ArrowDirection):
        super().__init__(coordinate, color)
        self.arrow_direction = arrow_direction

    def __repr__(self):
        return f"ArrowTile(coordinate={self.coordinate}, color={self.color}, arrow_direction={self.arrow_direction})"



